packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1505,6 +1507,17 @@ export async function renderToHTMLOrFlight(
           }
           if (isRedirectError(err)) {
             res.statusCode = 307
+            if (err.mutableCookies) {
+              const headers = getMutableCookieHeaders(
+                new Headers(),
+                err.mutableCookies
+              )
+              for (let [key, value] of headers) {
+                if (!forbiddenHeaders.includes(key)) {
```

Comment:
I wasn't sure if we need to exclude forbidden headers but I saw this was added in #49483 so I did the same here.

cc @ijjk

Reply:
I'm pretty sure we don't need `forbiddenHeaders` here so I removed it